test/distributed/test_nvshmem.py

```diff
@@ -637,10 +647,10 @@ def wait_until_kernel(
         out = symm_mem.empty(numel, dtype=dtype, device=self.device).fill_(-1)
         inp_hdl = symm_mem.rendezvous(inp, group=group_name)
         out_hdl = symm_mem.rendezvous(out, group=group_name)
-        dist.barrier()
 
         peer = 1 - rank
         NVSHMEM_CMP_EQ = 0  # from nvshmem.h
+        NVSHMEM_SIGNAL_SET = 0  # atomic set operation
```

Comment:
I see we can also pass in `NVSHMEM_SIGNAL_ADD` here? (https://docs.nvidia.com/nvshmem/api/gen/api/signal.html#available-signal-operators )

Do we see any scenario where it could be useful? @kwen2501

Reply:
yes, when the sender wants to monotonically update a counter as versioning info to the receiver.